### DISCUSSION TITLE
Prevent rerender by change event

### DISCUSSION THREAD

**Discussion by cakctus** [reduxjs/redux-toolkit#3047] [Q&A]
I have a state in my redux-toolkit, and this data changed by onChenge event, when i select something in my html list, and using this data in my rtk-query, but when my data in redux-toolkit changing also calling my useQuery, and i wanna callit by click event not by onChange event

**Comment by phryneas** [comment]
I am sorry, but without seeing any code it's impossible to give you any feedback on this.

**Reply by cakctus** [reply]
slice: 
![Screenshot_1](https://user-images.githubusercontent.com/39764706/209983220-970c8dc3-21aa-499f-a000-de8048fb477d.png)
api:
![Screenshot_2](https://user-images.githubusercontent.com/39764706/209983242-9b6b6c3e-2626-43d3-9916-69d69eb6166e.png)
call api:
![Screenshot_3](https://user-images.githubusercontent.com/39764706/209983274-4b0465a6-a615-4f89-93e6-2a1758894666.png)
change redux data by onChange event:
![Screenshot_4](https://user-images.githubusercontent.com/39764706/209983328-88efd301-44fc-44b0-a66d-4b9d077ebf4b.png)
call api by click:
![Screenshot_6](https://user-images.githubusercontent.com/39764706/209983398-5df6cf1b-45b8-49d0-bcc1-b308a59d61d7.png)

my useGetBodySearchResult is calling every time when onChange event is running, but I dont want this

hopefully does make sense

**Reply by phryneas** [reply]
Well, it will run every time you give it a new argument. If you do not want to run, maybe not give it a new argument? As long as you don't pass different search parameters in, it will not search for the new ones.

**Reply by cakctus** [reply]
Thanks for answer, its work,  but I give the skip second argument, in  useQuery, and should conditionally fetching,  no?

**Reply by phryneas** [reply]
Passing in `skip: true` will mean that you will not have any response available at all. From your code it's not visible if `skip` is `true` or `false` though.

**Comment by cakctus** [comment]
https://user-images.githubusercontent.com/39764706/209984066-e3a02ac1-3c3c-4011-968f-ca1e3e674641.mp4
